src/test/java/org/stellar/sdk/Sep10ChallengeTest.java

```diff
@@ -703,12 +740,35 @@ public void testReadChallengeTransactionInvalidDataValueWrongEncodedLength() thr
       Sep10Challenge.readChallengeTransaction(challenge, server.getAccountId(), Network.TESTNET, domainName);
       fail();
     } catch (InvalidSep10ChallengeException e) {
-      assertEquals("Random nonce encoded as base64 should be 64 bytes long.", e.getMessage());
+      assertEquals("Random nonce before encoding as base64 should be 48 bytes long.", e.getMessage());
     }
   }
 
   @Test
-  public void testReadChallengeTransactionInvalidDataValueCorruptBase64() throws IOException {
+  public void testReadChallengeTransactionValidDoesNotVerifyHomeDomain() throws IOException, InvalidSep10ChallengeException {
+    KeyPair server = KeyPair.random();
+    KeyPair client = KeyPair.random();
+    Network network = Network.TESTNET;
+    String domainName = "example.com";
+
+    long now = System.currentTimeMillis() / 1000L;
+    long end = now + 300;
+    TimeBounds timeBounds = new TimeBounds(now, end);
+
+    Transaction transaction = Sep10Challenge.newChallenge(
+        server,
+        network,
+        client.getAccountId(),
+        domainName,
+        timeBounds
+    );
+
+    Sep10Challenge.ChallengeTransaction challengeTransaction = Sep10Challenge.readChallengeTransaction(transaction.toEnvelopeXdrBase64(), server.getAccountId(), Network.TESTNET, null);
```

Comment:
Could you add a second test that is similar to this but passes in a value for home domain that is invalid? It's important that we test that if  the home domain is passed it is still not verified.

Reply:
Thanks, I have added the tests.